### PR TITLE
[HIBIKE] setup.sh update to include pip and pyserial

### PR DESCRIPTION
**Pull request by tingtingchiang**
Setup.sh for hibike now includes a check and install for python3-pip and pyserial which are needed for hibike_process.py. Had a new Ubuntu machine that did not have these modules installed yet.

### CODE REVIEW

**Comment by nikitakit** [comment]
Installing pip (if it's missing) looks good to me.

As for installing pyserial, I think using `requirements.txt` would be more ideal. This would also allow runtime's requirements.txt to pull in hibike changes. The overall diff would look something like:

```
diff --git a/hibike/requirements.txt b/hibike/requirements.txt
new file mode 100644
index 0000000..e6f6b6e
--- /dev/null
+++ b/hibike/requirements.txt
@@ -0,0 +1 @@
+pyserial==3.2.1
diff --git a/runtime/testy/requirements.txt b/runtime/testy/requirements.txt
index d69b7a4..910baf3 100644
--- a/runtime/testy/requirements.txt
+++ b/runtime/testy/requirements.txt
@@ -1,2 +1,2 @@
+-r ../../hibike/requirements.txt
 protobuf==3.2.0
-pyserial==3.2.1
```

(and also modifying `setup.sh` to install using the `requirements.txt` file)

File: hibike/setup.sh [hunk 1]
```diff
@@ -37,5 +37,14 @@ if [ "$(dpkg-query -W -f='${Status}' avr-libc 2>/dev/null | grep -c "ok installe
...
+	apt-get --assume-yes install python3-pip
+fi
+
+if [ "$(pip3 freeze | grep -c "pyserial")" != 1]; then
+        pip3 install pyserial
+fi
+
 
 echo "All dependencies installed."
+
```
Comment (yizow) [inline hibike/setup.sh]: remove extra newline

**Review by yizow — CHANGES_REQUESTED** [review]
Very minor newline change, else LGTM

File: runtime/testy/requirements.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
 protobuf==3.2.0
-pyserial==3.2.1
+-r ../../hibike/requirements.txt
+
```
Comment (kevinji) [inline runtime/testy/requirements.txt]: Stray newline here.

File: hibike/setup.sh [hunk 3]
```diff
@@ -37,5 +37,13 @@ if [ "$(dpkg-query -W -f='${Status}' avr-libc 2>/dev/null | grep -c "ok installe
 	apt-get --assume-yes install avr-libc	
 fi
 
+if [ "$(dpkg-query -W -f='${Status}' python-pip 2>/dev/null | grep -c "ok installed")" != 1 ]; then
+	apt-get --assume-yes install python3-pip
+fi
+
+if [ "$(pip3 freeze | grep -c "pyserial")" != 1]; then
+        pip3 install -r requirements.txt
```
Comment (kevinji) [inline hibike/setup.sh]: I think these are spaces and not tabs?

**Review by kevinji — CHANGES_REQUESTED** [review]
Minor spacing issues.

File: hibike/setup.sh (hunk 3 above)
Comment (yizow) [reply hibike/setup.sh]: I have no idea how you even catch these errors. GIVE ME YOUR VIMRC

Good catch

Comment (kevinji) [reply hibike/setup.sh]: This line and the `apt-get` line above looked off by one space so I was wondering why... then I tried highlighting the indentation.

**Review by yizow — APPROVED** [review]
LGTM

**Review by kevinji — APPROVED** [review]

File: hibike/setup.sh [hunk 4]
```diff
@@ -1,40 +1,48 @@
...
-	apt-get --assume-yes install arduino-mk	
+    apt-get --assume-yes install arduino-mk	
 fi
 
 if [ "$(dpkg-query -W -f='${Status}' avr-libc 2>/dev/null | grep -c "ok installed")" != 1 ]; then
-	apt-get --assume-yes install avr-libc	
+    apt-get --assume-yes install avr-libc	
+fi
+
+if [ "$(dpkg-query -W -f='${Status}' python-pip 2>/dev/null | grep -c "ok installed")" != 1 ]; then
```
Comment (kevinji) [inline hibike/setup.sh]: Should this be `python3-pip` instead of `python-pip`?

**Pull request merged**